collect_app/src/main/java/org/odk/collect/android/map/OsmMapFragment.java

```diff
@@ -338,37 +370,135 @@ protected void addMapLayoutChangeListener(MapView map) {
         return new MapPoint(geoPoint.getLatitude(), geoPoint.getLongitude(), geoPoint.getAltitude());
     }
 
-    protected static @NonNull MapPoint fromGeoPoint(@NonNull GeoPoint geoPoint, double sd) {
-        return new MapPoint(geoPoint.getLatitude(), geoPoint.getLongitude(), geoPoint.getAltitude(), sd);
+    protected static @NonNull MapPoint fromMarker(@NonNull Marker marker) {
+        GeoPoint geoPoint = marker.getPosition();
+        double sd = 0;
+        try {
+            sd = Double.parseDouble(marker.getSubDescription());
+        } catch (NumberFormatException e) {
+            Timber.w("Assertion violated: Marker.getSubDescription() did not contain a number");
```

Comment:
Same nit about the word assertion, maybe just say `"Error: ..."`?

Reply:
Done.